Assets/Modules/Movement/Scripts/PlayerMovementController.cs

```diff
@@ -68,6 +79,17 @@ private void OnJump(InputValue value)
         {
             aim.y = 1f;
         }
-        rb.AddForce(aim * jumpForce);
+        if (canJump)
+        {
+            rb.velocity = Vector2.zero;
+            rb.AddForce(aim * jumpForce);
+            canJump = false;
+        }
     }
+
+    private void OnCollisionEnter2D(Collision2D other)
+    {
+        canJump = true;
```

Comment:
Does this work without setting the `rb.velocity` to zero?

Reply:
the jump works, but it's that thing I mentioned before, that if I don't set it up to zero before jumping again and try to jump the opposite direction, rather than full jump to the other side, the forces are sort of cancelling each other out rather than jumping? putting it to zero fixes that issue.